server/src/main/java/org/elasticsearch/common/io/stream/RecyclerBytesStreamOutput.java

```diff
@@ -59,15 +64,27 @@ public long position() {
     @Override
     public void writeByte(byte b) {
         int currentPageOffset = this.currentPageOffset;
-        if (1 > (pageSize - currentPageOffset)) {
+        if (1 > pageSize - currentPageOffset) {
             ensureCapacity(1);
+            nextPage();
```

Comment:
Change because ensureCapacity now JUST allocates sufficient pages. Does not adjust write location.